### DISCUSSION TITLE
How can I customize my website footer

### DISCUSSION THREAD

**Discussion by ErikQQY** [facebook/docusaurus#4999] [Q&A] [answered]
Hello, I am trying to customize the footer of my website, like add some scripts and more pictures. I know that in Docusaurus v1, we can edit the Footer.js in ```website\core``` folder to customize the footer, but how can we do this in Docusaurus v2?

For example, I want to add GitHub stars in footer like Docusaurus v1, to realize the feature, all I need to do is add the script to ```Footer.js``` , how can we realize that feature in Docusaurus v2?

![屏幕截图 2021-06-18 120521](https://user-images.githubusercontent.com/52615090/122504822-a37e4800-d02d-11eb-9e5f-70bfc8c6d3b6.png)

**Comment by slorber** [comment, marked as answer]
You can add html elements in the footer, cf the docusaurus site itself:

```js

{
          title: 'More',
          items: [
            {
              label: 'Blog',
              to: 'blog',
            },
            {
              label: 'GitHub',
              href: 'https://github.com/facebook/docusaurus',
            },
            {
              label: 'Twitter',
              href: 'https://twitter.com/docusaurus',
            },
            {
              html: `
                <a href="https://www.netlify.com" target="_blank" rel="noreferrer noopener" aria-label="Deploys by Netlify">
                  <img src="https://www.netlify.com/img/global/badges/netlify-color-accent.svg" alt="Deploys by Netlify" />
                </a>
              `,
            },
          ],
        },
```

![image](https://user-images.githubusercontent.com/749374/122531838-30baa000-d020-11eb-943e-97a006800191.png)

Additional scripts can be added through config or plugin.

If you need more customization, you can swizzle the theme footer, you'll get a Footer.js file that you'll be able to modify as in v1

Note: Docusaurus v2 is an SPA (unlike v1). The integration of scripts might be a bit different as the footer may unmount/remount when navigating (current technical limitation, hard to solve). You may need to retrigger the script after each navigation, or call it in React useEffect.

**Comment by enthusiasticfolk** [comment]
Does this work on all CMSs? I recently collaborated with **Sturdy Business** for their [web design](https://sturdybusiness.com/web-design-importance-key-factors/), and I want to use the best resources to satisfy their clients.
